src/components/main/Note.jsx

```diff
@@ -0,0 +1,143 @@
+import { useNote, useAuth } from 'context/index'
+import axios from 'axios'
+import { useState, useReducer } from 'react'
+import { createNoteReducer } from 'reducer/createNoteReducer'
+import { colors } from './colors'
+
+export const Note = note => {
+  const {
+    note: { _id, title, body, color },
+  } = note
+  const { setNotes } = useNote()
+  const { encodedToken } = useAuth()
+  const [openEdit, setOpenEdit] = useState(false)
+  const [showPalette, setShowPalette] = useState(false)
+  const [editNote, editNoteDispatch] = useReducer(createNoteReducer, {
+    _id: _id,
+    title: title,
+    body: body,
+    color: color,
+  })
+  const deleteNoteHandler = async e => {
+    e.preventDefault()
+    try {
+      const response = await axios.delete(`/api/notes/${_id}`, {
+        headers: { authorization: encodedToken },
+      })
+      console.log(response)
+      setNotes(response.data.notes)
+    } catch (err) {
+      console.log(err)
+    }
+  }
+
+  const editNoteHandler = async e => {
+    e.preventDefault()
+    try {
+      const response = await axios.post(
+        `/api/notes/${_id}`,
+        {
+          note: editNote,
+        },
+        {
+          headers: { authorization: encodedToken },
+        }
+      )
+      console.log(response)
+      setOpenEdit(false)
+      setNotes(response.data.notes)
+    } catch (err) {
+      console.log(err)
+    }
+  }
+
+  const allFieldsAreFilled = editNote.title !== '' && editNote.body !== ''
+
+  return (
+    <div>
+      {openEdit ? (
+        <div className="note" style={{ backgroundColor: editNote.color }}>
+          <input
+            type="text"
+            placeholder="Title"
+            className="input__title"
+            value={editNote.title}
+            onChange={e =>
+              editNoteDispatch({ type: 'TITLE', payload: e.target.value })
+            }
+          />
+          <textarea
+            cols="30"
+            rows="10"
+            placeholder="Body of the Note"
+            className="input__body"
+            value={editNote.body}
+            onChange={e =>
+              editNoteDispatch({ type: 'BODY', payload: e.target.value })
+            }></textarea>
+          <div className="text__lg note__bottom">
+            <i
+              className="fa-solid fa-palette input__icons palette__container"
+              onClick={() => setShowPalette(!showPalette)}></i>
+            {showPalette && (
```

Comment:
You can make component for this palette.

Reply:
okay will do it